src/Raven.Server/Documents/ShardedHandlers/ShardedSmugglerHandler.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Smuggler;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.ShardedHandlers.ShardedCommands;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents;
+using Raven.Server.Smuggler.Documents.Data;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.ShardedHandlers
+{
+    public class ShardedSmugglerHandler : ShardedRequestHandler
+    {
+        [RavenShardedAction("/databases/*/smuggler/export", "POST")]
+        public async Task PostExport()
+        {
+            var result = new SmugglerResult();
+
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var operationId = GetLongQueryString("operationId", false) ?? ServerStore.Operations.GetNextOperationId();
+
+                try
+                {
+                    await Export(context, ShardedContext.DatabaseName, (options, blittableOptions, startDocumentEtag, startRaftIndex, onProgress, _, token) =>
+                            ExportShardedDatabaseInternalAsync(options, blittableOptions, startDocumentEtag, startRaftIndex, onProgress, context, token),
+                        ServerStore.Operations, operationId);
+                }
+                catch (Exception e)
+                {
+                    if (Logger.IsOperationsEnabled)
+                        Logger.Operations("Export failed .", e);
+
+                    result.AddError($"Error occurred during export. Exception: {e}");
+                    await WriteResultAsync(context, result, ResponseBodyStream());
```

Comment:
this will flush? I'm wondering if abort will not kill it brutally without flushing